packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -720,6 +721,46 @@ export class CdkToolkit {
     // If there is an '--app' argument and an environment looks like a glob, we
     // select the environments from the app. Otherwise, use what the user said.
 
+    const environments = await this.defineEnvironments(userEnvironmentSpecs);
+
+    await Promise.all(environments.map(async (environment) => {
+      success(' ⏳  Bootstrapping environment %s...', chalk.blue(environment.name));
+      try {
+        const result = await bootstrapper.bootstrapEnvironment(environment, this.props.sdkProvider, options);
+        const message = result.noOp
+          ? ' ✅  Environment %s bootstrapped (no changes).'
+          : ' ✅  Environment %s bootstrapped.';
+        success(message, chalk.blue(environment.name));
+      } catch (e) {
+        error(' ❌  Environment %s failed bootstrapping: %s', chalk.blue(environment.name), e);
+        throw e;
+      }
+    }));
+  }
+
+  /**
+   * Garbage collects assets from a CDK app's environment
+   * @param options Options for Garbage Collection
+   */
+  public async garbageCollect(userEnvironmentSpecs: string[], options: GarbageCollectionOptions) {
+    const environments = await this.defineEnvironments(userEnvironmentSpecs);
+
+    await Promise.all(environments.map(async (environment) => {
```

Comment:
Really, in parallel? If we do these in parallel that knocks other parallelism restrictions out of the water.

At least limit their parallelism, and think that if we do 10 of these in parallel, plus 50 of the tags, that means we can do 500 tags/second in total.

Reply:
i think i dont want this to be done in parallel at all. can revisit later.